packages/extension-browser/src/content-script/connector.ts

```diff
@@ -53,14 +53,15 @@ export default class WebExtensionConnector implements IConnector {
         });
 
         browser.runtime.onMessage.addListener(async (events: Events) => {
+
             try {
                 if (this._fetcher.handle(events)) {
                     return;
                 }
                 if (events.fetchEnd) {
                     await this.notifyFetch(events.fetchEnd);
                 }
-                if (events.fetchStart) {
+                if (events.fetchStart && !this.isExtensionResource(events.fetchStart.resource)) {
```

Comment:
Since we need to do this for both fetchStart and fetchEnd, I think I'd prefer an early exit check here like we did for `this._fetcher.handle` and just pass the `events` object to the helper.

Reply:
See if you like it better now.